application/src/main/java/org/thingsboard/server/service/notification/rule/DefaultNotificationRuleProcessingService.java

```diff
@@ -0,0 +1,237 @@
+        }
+
+        processTrigger(tenantId, triggerType, ruleEngineMsg.getOriginator(), ruleEngineMsg);
+    }
+
+    @Override
+    public void process(TenantId tenantId, Alarm alarm, boolean deleted) {
+        AlarmTriggerObject triggerObject = AlarmTriggerObject.builder()
+                .alarm(alarm)
+                .deleted(deleted)
+                .build();
+        processTrigger(tenantId, NotificationRuleTriggerType.ALARM, alarm.getId(), triggerObject);
+    }
+
+    @Override
+    public void process(TenantId tenantId, RuleChainId ruleChainId, String ruleChainName, EntityId componentId, String componentName, ComponentLifecycleEvent eventType, Exception error) {
+        RuleEngineComponentLifecycleEventTriggerObject triggerObject = RuleEngineComponentLifecycleEventTriggerObject.builder()
+                .ruleChainId(ruleChainId)
+                .ruleChainName(ruleChainName)
+                .componentId(componentId)
+                .componentName(componentName)
+                .eventType(eventType)
+                .error(error)
+                .build();
+        processTrigger(tenantId, NotificationRuleTriggerType.RULE_ENGINE_COMPONENT_LIFECYCLE_EVENT, componentId, triggerObject);
+    }
+
+    private void processTrigger(TenantId tenantId, NotificationRuleTriggerType triggerType, EntityId originatorEntityId, Object triggerObject) {
+        ListenableFuture<List<NotificationRule>> rulesFuture = dbCallbackExecutor.submit(() -> {
+            return notificationRuleService.findNotificationRulesByTenantIdAndTriggerType(tenantId, triggerType);
+        });
+        DonAsynchron.withCallback(rulesFuture, rules -> {
+            for (NotificationRule rule : rules) {
+                notificationExecutor.submit(() -> {
+                    processNotificationRule(rule, originatorEntityId, triggerObject);
+                });
+            }
+        }, e -> {
+            log.error("Failed to find notification rules by trigger type {}", triggerType, e);
+        });
+    }
+
+    private void processNotificationRule(NotificationRule rule, EntityId originatorEntityId, Object triggerObject) {
+        NotificationRuleTriggerConfig triggerConfig = rule.getTriggerConfig();
+        log.debug("Processing notification rule '{}' for trigger type {}", rule.getName(), rule.getTriggerType());
+
+        if (triggerConfig.getTriggerType().isUpdatable()) {
+            List<NotificationRequest> notificationRequests = notificationRequestService.findNotificationRequestsByRuleIdAndOriginatorEntityId(rule.getTenantId(), rule.getId(), originatorEntityId);
+            if (!notificationRequests.isEmpty()) {
+                if (matchesClearRule(triggerObject, triggerConfig)) {
+                    notificationRequests = notificationRequests.stream()
+                            .filter(notificationRequest -> {
+                                if (!notificationRequest.isSent()) {
+                                    dbCallbackExecutor.submit(() -> {
```

Comment:
dbCallbackExecutor is used to process callbacks and not to execute DB call requests. It is either wrong naming or wrong executor used.